src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1454,7 +1454,17 @@ private BoundExpression BindDefaultExpression(DefaultExpressionSyntax node, Bind
             TypeWithAnnotations typeWithAnnotations = this.BindType(node.Type, diagnostics, out AliasSymbol alias);
             var typeExpression = new BoundTypeExpression(node.Type, aliasOpt: alias, typeWithAnnotations);
             TypeSymbol type = typeWithAnnotations.Type;
-            return new BoundDefaultExpression(node, typeExpression, constantValueOpt: type.GetDefaultValue(), type);
+
+            bool hasErrors = false;
+
+            if (this.InAttributeArgument && type.IsEnumType() && type.ContainsFunctionPointer())
```

Comment:
>type.IsEnumType [](http://example.com/codeflow?start=44&length=15)

Could you add a comment explaining what is special about an enum here?

Reply:
Yes, added.